os/audio/audio.c

```diff
@@ -693,6 +693,42 @@ static inline void audio_dequeuebuffer(FAR struct audio_upperhalf_s *upper, FAR
 	}
 }
 
+/****************************************************************************
+ * Name: audio_handleerror
+ *
+ * Description:
+ *   Send an AUDIO_MSG_IOERR message to the client to indicate that the
+ *   an error has occured.  The lower-half driver initiates this
+ *   call via its callback pointer to our upper-half driver.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_AUDIO_MULTI_SESSION
+static inline void audio_handleerror(FAR struct audio_upperhalf_s *upper, FAR struct ap_buffer_s *apb, uint16_t status, FAR void *session)
+#else
+static inline void audio_handleerror(FAR struct audio_upperhalf_s *upper, FAR struct ap_buffer_s *apb, uint16_t status)
```

Comment:
you should check status.
please use switch-case for it.

Reply:
Since we have only underrun and overrun errors, I am not checking the status. 
I am always sending XRUN msg and it is implicit that overrun during record and underrun during playback. 
In future, if we have some new error types, we can check status and pass msg accordingly. But, I am not expecting any new error msgs.